packages/core/addon/components/data-availability.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2022, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { later, cancel } from '@ember/runloop';
+import { helper } from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import { Status } from 'navi-data/services/data-availability';
+import { getDataSource } from 'navi-data/utils/adapter';
+import type DataAvailabilityService from 'navi-data/services/data-availability';
+import type { DataSourceAvailabilty, AvailabilityResult } from 'navi-data/services/data-availability';
+import type { TaskGenerator, TaskInstance } from 'ember-concurrency';
+import { sortBy } from 'lodash-es';
+
+type Timer = ReturnType<typeof later>;
+//@ts-ignore - TODO: cannot import Dropdown type
+type Dropdown = any;
+
+interface DataAvailabilityArgs {
+  dataSources?: string[];
+}
+
+type DataSourceStatus = { dataSource: string; displayName: string; result: AvailabilityResult };
+type AllAvailabilityStatus = { dataSources: DataSourceStatus[]; status: Status };
+
+const REFRESH_POLL_MINUTES = 10;
+const TABBED_OUT_POLL_MINUTES = 60;
+
+export function overallAvailability(availabilities: DataSourceStatus[]) {
+  const statuses = new Set(availabilities.map((v) => v.result.status));
+  if (statuses.size === 0) {
+    return Status.UNAVAILABLE;
+  } else if (statuses.size === 1) {
+    return statuses.keys().next().value;
+  }
+  return Status.DELAYED;
+}
+
+export default class DataAvailability extends Component<DataAvailabilityArgs> {
+  @tracked declare availabilities: TaskInstance<AllAvailabilityStatus>;
+
+  @service
+  declare dataAvailability: DataAvailabilityService;
+
+  isHidden = false;
+
+  constructor(owner: unknown, args: DataAvailabilityArgs) {
+    super(owner, args);
+    document.addEventListener('visibilitychange', () => {
```

Comment:
do we need to remove this listener on `will-destroy`?

Reply:
Seems like we are doing more of this. A service could be nice.